osf/models/sanctions.py

```diff
@@ -167,6 +186,23 @@ def _on_complete(self, user):
     def forcibly_reject(self):
         self.state = Sanction.REJECTED
 
+    def _save_transition(self, event_data):
+        """Recored the effects of a state transition in the database."""
+        self.save()
+        new_state = event_data.transition.dest
+        # No need to update registration state with no sanction state change
+        if new_state is None:
+            return
+
+        user = event_data.kwargs.get('user')
+        if user is None and event_data.kwargs:
```

Comment:
```suggestion
        if user is None and event_data.args:
```
i assume that's the intent, since the if's block uses `event_data.args[0]`